### DISCUSSION TITLE
an-introduction-to-primary-constructors-in-csharp-12/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#419] [Announcements]
# an-introduction-to-primary-constructors-in-csharp-12/

In this post I introduce C#12 primary constructors, describe the various ways to use them and how they work behind the scenes.

https://andrewlock.net/an-introduction-to-primary-constructors-in-csharp-12/

**Comment by NijoLabs** [comment]
Great article, explained the concept perfectly. I hope they add readonly support so that it can be used in more scenarios.

**Comment by gao-artur** [comment]
I wish the never release this feature. It's great for POCO's but a mess for regular classes.

**Comment by marc-lewandowski** [comment]
Would be great if you could cover the `in` parameter decoration as well.
